modules/trusted-firmware-m/Kconfig.tfm

```diff
@@ -267,10 +280,10 @@ config TFM_REGRESSION_S
 	  the Secure domain regression tests.
 
 config TFM_REGRESSION_NS
-	bool "Use the TF-M Non-Secure Regression test application"
+	bool "TF-M Non-Secure Regression tests"
 	help
-	  When this is enabled, the Zephyr application as a whole will be
-	  replaced with the TF-M Non-Secure Regression test application.
+	  When enabled, this option signifies that the TF-M build includes
+	  the Non-Secure domain regression tests.
```

Comment:
There is no dependency to `TFM_APP_NS` meaning as a user I can do:
`TFM_APP_NS=n` and `TFM_REGRESSION_NS=y` in which case I believe that `zephyr.hex` will be used in the merged hex and not the regression test non-secure hex file.

Not sure the exact use-case for building the regression tests, but not including them in anything ?

Ref: 
https://github.com/zephyrproject-rtos/zephyr/blob/1171a23e11825e480221de8b2caa2bd7d4a15acd/modules/trusted-firmware-m/CMakeLists.txt#L470-L476

Reply:
I don't have any use-cases for doing it, but in theory this might be possible.
```
TFM_BUILD_NS=y
TFM_APP_NS=n
TFM_REGRESSION_TEST=y
```

The application might want to link in something here:
https://github.com/zephyrproject-rtos/zephyr/blob/1171a23e11825e480221de8b2caa2bd7d4a15acd/modules/trusted-firmware-m/CMakeLists.txt#L397-L402

No objection to adding a depends on though, just saying that there really isn't a need for a dependency here.
The depends would just be to disallow a configuration then.

Reply:
Already today TF-M is a complex system, and by reducing number of settings to consider, either because they are dedicated in special (test) samples or unavailable because `depends on` I believe we make it a little easier for developers.
Reduce decision stress.

For `TFM_REGRESSION_NS=y` I don't directly see why we would want to link those regression tests into a real sample.
To my understanding those are independent tests to be run independent of any application, so I believe just providing the ability to enable this when linking directly to the TF-M ns app will hide such settings from users.

This actually made me take a higher level look at the TF-M Kconfig, and maybe we should have a dedicated TF-M test configuration part, so that it is clearer when TF-M tests are built, and when you might built an application.
But such refactoring is outside scope of this PR.

Reply:
Yeah, refactoring is a bit much for this PR. But I have updated the description so that it is clear that those are meant for testing purposes and shouldn't be used by the user.